comid/integrityregisters.go

```diff
@@ -136,3 +137,21 @@ func (i *IntegrityRegisters) UnmarshalJSON(data []byte) error {
 	}
 	return nil
 }
+
+func (i *IntegrityRegisters) Compare(r *IntegrityRegisters) error {
+	claim, err := i.MarshalJSON()
```

Comment:
Same comment as above, and in this case, since `IntegrityRegisters` has a map, byte comparison of JSON is _definitely_ not safe.